### PR TITLE
Add missing auto update feature

### LINKED ISSUE #833: Add dropped auto-update feature
With PR https://github.com/kubermatic/kubermatic/pull/748 we dropped the update-controller.
With that we also dropped the feature of "auto-updates"

After merge of https://github.com/kubermatic/kubermatic/pull/748 there is no need to have a dedicated update-controller. But we need to have a logic which auto-updates clusters when specified.

Definition "auto-update":
The controller has a "versions" file containing all supported versions (kubernetes version, etcd, machine-controller, etc...).
Also the controller has a "updates" file describing possible update paths. (1.8.4 -> 1.8.5, 1.8.5 -> 1.9.0). A update path can also be marked as `automatic: true` meaning that all clusters matching a specific version will be automatically updated to the specified version.

In the current state, the controller only needs to update the masterVersion field on the cluster spec.
The rest will be taken care of automatically.

### PR DESCRIPTION
**Pull request by mrIncompetent**
**What this PR does / why we need it**:
Enables automatic updates based on the definitions in the updates.yaml.

The controller will automatically update all manifests.

Fixes #833

### CODE REVIEW

File: api/pkg/controller/cluster/pending.go [hunk 1]
```diff
@@ -202,6 +207,20 @@ func (cc *Controller) ensureCloudProviderIsInitialize(cluster *kubermaticv1.Clus
 	return nil
 }
 
+func (cc *Controller) ensureAutomaticMasterUpdate(c *kubermaticv1.Cluster) error {
+	updateVersion, err := version.BestAutomaticUpdate(c.Spec.MasterVersion, cc.updates)
+	if err != nil {
+		return err
+	}
+
+	if updateVersion != nil {
```
Comment (p0lyn0mial) [inline api/pkg/controller/cluster/pending.go:216]: does nil `updateVersion` mean that automatic upgrade is not possible ?

Comment (mrIncompetent) [reply api/pkg/controller/cluster/pending.go:216]: `nil` in this case means, there is no update to apply automatically

Comment (p0lyn0mial) [reply api/pkg/controller/cluster/pending.go:216]: should we return an error is such a case ?

Comment (mrIncompetent) [reply api/pkg/controller/cluster/pending.go:216]: No, its fine to not have an automatic update available.
We will run into this case for every cluster which is already on the latest version

**Review by p0lyn0mial — APPROVED** [review]

**Pull request merged**